onnxruntime/core/providers/cuda/tensor/compress.cc

```diff
@@ -36,7 +36,8 @@ Status Compress::ComputeInternal(OpKernelContext* ctx) const {
   int64_t compress_input_length = has_axis_ ? input_dimensions[axis_] : input_size;
   int64_t valid_condition_length = compress_input_length < condition_length ? compress_input_length : condition_length;
 
-  auto condition_cumulative_sum = GetScratchBuffer<int32_t>(valid_condition_length).get();
+  auto condition_cumulative_sum_buffer = GetScratchBuffer<int32_t>(valid_condition_length);
+  auto condition_cumulative_sum = condition_cumulative_sum_buffer.get();
```

Comment:
>condition_cumulative_sum_buffer [](start = 34, length = 31)

Thanks for the fix. Do we know why it was not caught by tests? How can we prevent future errors like this?